### PR TITLE
Add build custom base image functionality

### PR DESCRIPTION
**Pull request by lcarva**
Requires corresponding release of atomic-reactor with changes from: https://github.com/projectatomic/atomic-reactor/pull/493

### CODE REVIEW

File: osbs/build/build_request.py [hunk 1]
```diff
@@ -501,6 +513,41 @@ def adjust_for_scratch(self):
...
+            if not self.spec.kojihub.value:
+                raise OsbsValidationException(
+                    'Custom base image builds require kojihub to be defined')
+            self.dj.dock_json_set_arg(phase, plugin, 'koji_hub',
+                                      self.spec.kojihub.value)
+            if self.spec.proxy.value:
+                self.dj.dock_json_set_arg(phase, plugin, 'koji_proxyuser',
+                                          self.spec.proxy.value)
+
+
```
Comment (twaugh) [inline osbs/build/build_request.py]: Nitpick: PEP8 only needs one blank line between method definitions.

File: tests/build/test_build_request.py [hunk 2]
```diff
@@ -27,6 +27,45 @@
 class NoSuchPluginException(Exception):
     pass
```
Comment (twaugh) [inline tests/build/test_build_request.py]: Nitpick: PEP8 requires two blank lines between function definitions.

**Comment by twaugh** [comment]
LGTM

**Comment by lcarva** [comment]
Addressed PEP8 issues.

**Comment by mmilata** [comment]
LGTM, merged!

**Pull request merged**